packages/oui-numeric/src/numeric.controller.js

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash'
+
+// By design, value is restricted to [0, 99999] interval
+const MIN_VALUE = 0
+const MAX_VALUE = 99999
+
+export default class {
+  constructor ($attrs, $element, $log) {
+    'ngInject'
+
+    this.$attrs = $attrs
+    this.$element = $element
+    this.$log = $log
+  }
+
+  $postLink () {
+    this.$element.removeAttr('id')
+    this.$element.removeAttr('name')
+  }
+
+  $onInit () {
+    if (!this.id) {
+      this.$log.warn(`Missing required attribute id`)
```

Comment:
For checkbox if `id` is missing i'm generating one using `$scope.$id` like `oui-checkbox-0`. With that id will always be unique and `label for` will work without requiring the user to enter an id.